### PR TITLE
[WIP] add gitea platform

### PR DESCRIPTION
**Pull request by jgarec**
This PR Adds a new platform : gitea

URL : https://gitea.com
Sandbox : https://try.gitea.io
API : https://gitea.com/api/swagger

Firsts tests are ok : connect to the platform, clone repositories, create a PR and issue.
APIs are quite similar to gitlab API

### CODE REVIEW

**Comment by rarkins** [comment]
@jgarec sounds interesting!

> create a PR and issue

Not added to this branch yet though?

**Comment by qzmfranklin** [comment]
This is really interesting! Our team uses Gitea instead of github for improved network speed.

Would love to use renovate with Gitea!

File: lib/platform/gitea/index.ts [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+  if (endpoint) {
+    res.endpoint = endpoint.replace(/\/?$/, '/'); // always add a trailing slash
+    api.setBaseUrl(res.endpoint);
+    defaults.endpoint = res.endpoint;
+  } else {
+    res.endpoint = defaults.endpoint;
+    logger.info('Using default Gitea endpoint: ' + res.endpoint);
+  }
+  try {
+    res.gitAuthor = (await api.get(`user?token=${token}`)).body.email;
```
Comment (rarkins) [inline lib/platform/gitea/index.ts]: Let's go with the header-based approach for tokens:

image

File: lib/platform/gitea/index.ts [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+}
+
+// Get all repositories that the user has access to
+export async function getRepos() {
+  logger.info('Autodiscovering Gitea repositories');
+  try {
+    // TODO : check paginate capabilities
+    // TODO : limit to user repositories (add uid parameter ?)
+    // const url = `repos?membership=true&per_page=100`;
+    const url = `repos/search/`;
```
Comment (rarkins) [inline lib/platform/gitea/index.ts]: Does this return only repos that the authenticated user has access to? Or possibly all repos on the system?

**Comment by cleverer** [comment]
Is there any progress on this? Is there a plan when it will be released?

**Comment by rarkins** [comment]
This is a community-contributed PR so there’s no plan. It looks like it might need someone else to take it up

**Comment by jgarec** [comment]
Hi, I'm still planning to work on this PR... but i need an internet connection at home which is not the case since i moved 3 months ago ... 😥

**Comment by rarkins** [comment]
This has fallen too far behind so closing it. Happy for anyone to restart work on this in future and submit a new PR.

**Pull request closed without merging**